client/signup/steps/user/index.jsx

```diff
@@ -104,6 +104,8 @@ export class UserStep extends Component {
 			locale: this.props.locale,
 			oauth2ClientId: this.props.oauth2Client && this.props.oauth2Client.id,
 			wccomFrom: this.props.wccomFrom,
+			isWhiteLogin: this.props.isReskinned,
+			signupUrl: window.location.pathname + window.location.search,
 		} );
 	}
```

Comment:
I'm adding this comment here because as a user, the login link has a localization issue. But honestly, I have no idea what is causing this.

The problem is only when no locale param is provided in the URL and the browser main language is used to translate the UI of the signup form.

**Steps to reproduce:**
1. Set the main browser language to something else than EN.

<img width="919" alt="Screenshot 2021-08-04 at 16 47 50" src="https://user-images.githubusercontent.com/14192054/128194229-1cb9187e-52e2-4f2d-81b8-5cfbe0c57ab4.png">

2. Use the live link and go to `/start` while logged out

<img width="1206" alt="Screenshot 2021-08-04 at 16 49 09" src="https://user-images.githubusercontent.com/14192054/128195113-cde48aa7-40cd-4016-9e24-a62c9a1841e3.png">

3. The login URL doesn't contain the locale as expected so the login page won't be translated

**Expected behaviour:**
- reskin in production
<img width="1055" alt="Screenshot 2021-08-04 at 16 52 37" src="https://user-images.githubusercontent.com/14192054/128194529-15cc1c77-9a87-4a56-a23c-06d0413b09b3.png">

- blue signup on live link of this branch
<img width="1197" alt="Screenshot 2021-08-04 at 16 48 32" src="https://user-images.githubusercontent.com/14192054/128194591-cbebe43c-98ff-4a0d-927f-30cd9fc936e4.png">

**Additional note:**
- the locale is added correctly only in the case when it's present in the URL and not inferred from the browser settings.

<img width="1210" alt="Screenshot 2021-08-04 at 16 51 07" src="https://user-images.githubusercontent.com/14192054/128194840-0668087d-3b97-4b7e-aca2-cc1eef519353.png">

Reply:
Looks like this is caused by the wrong (previous) `locale` used in `getLoginLink` (and other places due to the issue I mentioned yesterday). I pinged you in #55201 where I do some cleanup. Also rebased this on top of #55201, so hopefully it's resolved in the last build. Mind checking again pls? 😃

Reply:
The fix is looking good for that case. Now we should do the same for this link:

<img width="469" alt="Screenshot 2021-08-11 at 15 10 27" src="https://user-images.githubusercontent.com/14192054/129027944-a706d532-3060-43dd-a4b0-de51514516b3.png">